### ISSUE TITLE
fatal: the file used by another process

### ISSUE THREAD

**Issue by marakew** [ninja-build/ninja#2416]
cmake + ninja 1.12.0 on Windows

>CreateProcess failed. Command attempted:
>.......
>CreateProcess: The process cannot access the file because it is being used by another process

is this normal ?

run build again and no issue

but it is strange

which the file ninja mean ?

iam sure no body use the file except ninja build

**Comment by jhasse** [comment]
Maybe Anti-Virus? Can you try with disabling the real time scanning?

**Comment by marakew** [comment]
I doubt

I've been using cmake +ninja for over a year

hardware and software have not changed

except that I updated ninja yesterday

I used 1.11.1 seems all was good

I will tray again at next week rebuild for scrach, for reproduce the bug

there are many stranges with 1.12.0 version

1) targets != builds targets

after done i see at log

> [1048/1049] cmd.exe ......

where is +1 build ?

i dont not stop/restart the build process

2) while ninja builds, it reorder the targets

is this ok ?

**Comment by jhasse** [comment]
> is this ok ?

Yes, this is a new feature: Based on the duration of each job, it will reorder them to maximize parallelism, so that the build finishes faster.

**Comment by marakew** [comment]
1) cmake dev says its ninja issues

now i got 

....
FAILED: CMakeFiles/prj.dir/src/processing.cpp.obj
....

without any explain why did this happen 

and i cant find any error fail in the whole log file

**Comment by jhasse** [comment]
Please try with Anti-Virus disabled. And also don't put other issues into this one, open new ones instead. It's hard to follow, sorry.

**Comment by marakew** [comment]
yes i see the command

but after start build again, same command ok

i think not all corner cases can handle FAILED with error

i mean compiler can terminate process with some exit error code, but ninja did not show it number

can FAILED be improve by show exit code number of compiler process ?

**Comment by marakew** [comment]
as i understand stderr was empty in this case

**Comment by marakew** [comment]
see above

i see the command

i can rerun it and ninja was rerun it when i start build again, and all was ok

but why this FAILED happen i cant see, i mean exist code of process which produce FAILED for this case

**Comment by marakew** [comment]
will see after long period, month or few month

i upgrate ninja few days ago

there are nee more statistics for understand is this issue or some thing else

**Comment by jhasse** [comment]
See #1507 regarding the exit code.
